python/rmm/tests/test_rmm.py

```diff
@@ -942,3 +943,46 @@ def test_rmm_device_buffer_copy(cuda_ary, make_copy):
     result = db_copy.copy_to_host()
 
     np.testing.assert_equal(expected, result)
+
+
+@pytest.mark.parametrize("level", rmm.logging_level)
+def test_valid_logging_level(level):
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            "ignore", message="RMM will not log logging_level.TRACE."
+        )
+        warnings.filterwarnings(
+            "ignore", message="RMM will not log logging_level.DEBUG."
+        )
+        rmm.set_logging_level(level)
+        assert rmm.get_logging_level() == level
+        rmm.set_logging_level(rmm.logging_level.INFO)  # reset to default
+
+
+invalid_logging_levels = ["INFO", 3, "invalid", 100, None, 1.2345, [1, 2, 3]]
+
+
+@pytest.mark.parametrize("level", invalid_logging_levels)
+def test_invalid_logging_level(level):
+    with pytest.raises(TypeError):
+        rmm.set_logging_level(level)
```

Comment:
suggestion (non-blocking): Is it worth testing other APIs too like `should_log` and `set_flush_level`? Since all of them are doing validation on their own?

Reply:
set_flush_level is tested below. But I can consolidate with fixtures

Reply:
I consolidated, and got rid of the global variable.